lib/vehicle_factory.rb

```diff
@@ -0,0 +1,21 @@
+class VehicleFactory 
+
+    def initialize 
+    end
+
+    def create_vehicles(registration_locations)
+        registration_locations.each do |vehicle|
+            vehicle_details = ({vin: vehicle[:vin_1_10] , year: vehicle[:model_year], make: vehicle[:make], model: vehicle[:model], engine: [:ev]})
+            new_vehicles = Vehicle.new(vehicle_details)
+            return new_vehicles
```

Comment:
This explicit return will only return this one vehicle that you're iterating through. We want to create a collection of vehicles and return the entire collection of vehicles. A placeholder might be helpful to shovel in each vehicle that you create, and then you'll need to return that placeholder after you're done iterating. 

```ruby 
vehicles = []
registration_locations.each do |vehicle|
            vehicle_details = ({vin: vehicle[:vin_1_10] , year: vehicle[:model_year], make: vehicle[:make], model: vehicle[:model], engine: [:ev]})
            vehicles << Vehicle.new(vehicle_details)
     end  
   vehicles
end 
```